src/task.js

```diff
@@ -0,0 +1,85 @@
+// To configure, define in your environment:
+//   OWNER_EMAIL
+//   TASKCLUSTER_ACCESS_TOKEN
+//   TASKCLUSTER_CLIENT_ID
+//
+// Notes:
+// * Only supports a workerType with permacreds at this time.
+
+export function generateCloneTaskDefinition(repo) {
+
+    var date = new Date();
+    var deadline = new Date(date.valueOf() + 3600 * 1000);
+    var params = ['create-clone-cache', '--upload', '--proxy', repo]
+
+    var task = {
```

Comment:
not that it's a huge issue, but because most of this task definition is the same between these methods, we could load it in from elsewhere and then do something like

Object.assign({}, task, { deadline: ..., created: ..., payload: { command: <.some command> } });